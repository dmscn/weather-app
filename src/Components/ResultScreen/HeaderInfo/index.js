export {default} from './HeaderInfo';
export * from './HeaderInfo';