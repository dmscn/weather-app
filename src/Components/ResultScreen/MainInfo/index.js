export {default} from './MainInfo';
export * from './MainInfo';