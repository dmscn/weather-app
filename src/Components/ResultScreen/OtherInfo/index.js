export {default} from './OtherInfo';
export * from './OtherInfo';