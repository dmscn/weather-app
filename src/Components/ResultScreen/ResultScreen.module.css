.Container {
  display: flex;
  width: 100vw;
  height: 100vh;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

hr {
  display: block;
  width: 70vw;
  margin: 2em 0;
}