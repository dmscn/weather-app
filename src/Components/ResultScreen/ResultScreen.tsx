import React, { Component } from 'react';
import moment from 'moment';

import styles from './ResultScreen.module.css';
import HeaderInfo from './HeaderInfo';
import MainInfo from './MainInfo';

moment.locale('pt-BR');
export interface Props {
  data: any,
}

export default class ResultScreen extends Component<Props> {
  render(): JSX.Element {
    const { data } = this.props;
    return (
      <main className={styles.Container}>
        <HeaderInfo location={data.name} />
        <MainInfo 
          weatherId={data.weather[0].id}
          weather={data.weather[0].main}
          temp={data.main.temp}
          tempMax={data.main.temp_max}
          tempMin={data.main.temp_min}
        />
      </main>
    )
  }
}
