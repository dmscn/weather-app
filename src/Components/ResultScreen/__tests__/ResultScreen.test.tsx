import React from 'react'
import { shallow } from "enzyme";
import ResultScreen from "../ResultScreen";

describe('<ResultScreen />', () => {
  it('renders', () => {
    const wrapper = shallow(<ResultScreen />);
    expect(wrapper).toMatchSnapshot();
  })
});