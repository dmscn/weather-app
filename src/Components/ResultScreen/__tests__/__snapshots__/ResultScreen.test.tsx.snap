// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultScreen /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultScreen
    data={
      Object {
        "name": "Rio de Janeiro",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <HeaderInfo
        location="Rio de Janeiro"
      />,
      "className": "Container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "location": "Rio de Janeiro",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <HeaderInfo
          location="Rio de Janeiro"
        />,
        "className": "Container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "location": "Rio de Janeiro",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
